cronjob_scripts/crates_io_popular_crates.py

```diff
@@ -0,0 +1,83 @@
+import subprocess
+import tarfile
+from tempfile import TemporaryDirectory
+from shutil import copyfileobj
+
+import requests
+import polars as pl
+
+class TarEntry:
+    def __init__(self, tarball: tarfile.TarFile, member: tarfile.TarInfo):
+        self.tarball = tarball
+        self.member = member
+
+    def is_file(self):
+        return self.member.isfile()
+
+    @property
+    def name(self):
+        return self.member.name
+
+    def extract_to(self, path: str):
+        with open(path, mode="xb") as f:
+            stream = self.tarball.extractfile(self.member)
+            assert stream
+            copyfileobj(stream, f)
+
+def download_tar_gz(url):
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+
+    with tarfile.open(fileobj=response.raw, mode='r:gz') as tarball:
+        while True:
+            member = tarball.next()
+            if member is None:
+                return
+            yield TarEntry(tarball, member)
+
+def get_crates_io_popular_crates(minimum_downloads=4000):
```

Comment:
I have no idea how many crates this will return. If it returns a lot of crates then our current scheme will take an awfully long time getting around to checking the most popular ones. Can we limit the total number to 100 or something? (not sure what a sensible number should be). See #284 

I was also thinking about whether we could pick a sampling algorithm that favours highly requested packages but still also checks rarely requested packages sometimes. Maybe multiply up each crate in the list by its request count, and then shuffle, and then deduplicate? There's probably a better way to do it though.

Reply:
We can tune the packages returned via `minimum_downloads`.

If the minimum_downloads is 4000, then it would return 929 crates.

If you increase it, then the number decrease very quickly.

You can play with that file a bit

> I was also thinking about whether we could pick a sampling algorithm that favours highly requested packages but still also checks rarely requested packages sometimes. 

IMO that is covered by our stats server.

Reply:
I've increased `minimum_downloads` to `200000`, so it only returns ~300 crates